Lib/test/test_httpservers.py

```diff
@@ -551,13 +552,14 @@ def test_undecodable_filename(self):
         self.check_status_and_reason(response, HTTPStatus.OK,
                                      data=os_helper.TESTFN_UNDECODABLE)
 
+    @unittest.skipIf(locale.getpreferredencoding().lower() != 'utf-8', 'test requires utf-8')
     def test_undecodable_parameter(self):
         # sanity check using a valid parameter
         response = self.request(self.base_url + '/?x=123').read()
-        self.assertRegex(response, rf'listing for {self.base_url}/\?x=123'.encode('latin1'))
-        # now the bogus encoding
+        self.assertRegex(response, rf'listing for {self.base_url}/\?x=123'.encode('utf-8'))
+        # now the bogus (utf-8) encoding
         response = self.request(self.base_url + '/?x=%bb').read()
-        self.assertRegex(response, rf'listing for {self.base_url}/\?x=\xef\xbf\xbd'.encode('latin1'))
+        self.assertRegex(response, rf'listing for {self.base_url}/\?x=\xef\xbf\xbd'.encode('utf-8'))
```

Comment:
I think we need to better understand what this test is doing before changing this.  Generally the `latin1` encoding is used in the test suite to convert `str` directly  to `bytes` without the risk of an encoding changing this, which can happen with `utf-8` in particular in some small range which I forget off the top of my head.

Maybe the skip can go ahead, but the encoding within the test should not change without a very good reason.